hummingbird/ml/operator_converters/_tree_implementations.py

```diff
@@ -329,17 +329,24 @@ def aggregation(self, x):
         return x
 
     def forward(self, x):
+        # TODO: obtain batch size at compile time
+        batch_size = x.shape[0]
+
         prev_indices = (torch.ge(torch.index_select(x, 1, self.root_nodes), self.root_biases)).long()
         prev_indices = prev_indices + self.tree_indices
-        prev_indices = prev_indices.view(-1)
+        # prev_indices = prev_indices.view(-1)
+        prev_indices = prev_indices
 
         factor = 2
         for nodes, biases in zip(self.nodes, self.biases):
-            gather_indices = torch.index_select(nodes, 0, prev_indices).view(-1, self.num_trees)
-            features = torch.gather(x, 1, gather_indices).view(-1)
-            prev_indices = factor * prev_indices + torch.ge(features, torch.index_select(biases, 0, prev_indices)).long()
-
-        output = torch.index_select(self.leaf_nodes, 0, prev_indices).view(-1, self.num_trees, self.n_classes)
+            # gather_indices = torch.index_select(nodes, 0, prev_indices).view(-1, self.num_trees)
```

Comment:
Can you please add some comments about the things you left commented out?  (else delete it if you think it will not be used again?)
Thanks!